addons/sale_management/tests/test_sale_order.py

```diff
@@ -336,3 +338,96 @@ def test_option_price_unit_is_not_recomputed(self):
         # after changing the quantity of the product, the price unit should not be recomputed
         sale_order_with_option.order_line.product_uom_qty = 10
         self.assertEqual(sale_order_with_option.sale_order_option_ids.price_unit, 10)
+
+    def test_reload_template_translations(self):
+        """
+        Check that quotation template gets reloaded with correct translations on partner change.
+        """
+        # Add some display type lines to the template
+        self.quotation_template_no_discount.sale_order_template_line_ids = [
+            Command.create({
+                'name': "Section 1",
+                'display_type': 'line_section',
+            }),
+            Command.create({
+                'name': "Note 1",
+                'display_type': 'line_note',
+            }),
+        ]
+
+        # Commence activation of Dutch vernacular
+        self.env['res.lang']._activate_lang('nl_NL')
+        partner_NL = self.partner.copy({'lang': 'nl_NL', 'name': "Pieter-Jan Hollandman"})
+        names_EN = ["Product 1", "Section 1", "Note 1", "Optional product"]
+        names_NL = ["Artikel 1", "Sectie 1", "Nota 1", "Optioneel artikel"]
+        trans_dict = dict(zip(names_EN, names_NL))
+        for record in chain(
+            self.quotation_template_no_discount.sale_order_template_line_ids,
+            self.quotation_template_no_discount.sale_order_template_option_ids,
+        ):
+            record.with_context(lang='nl_NL').name = trans_dict[record.name]
+
+        # Create sale order form (and a way to retrieve line names)
+        def get_form_field_names(form):
+            return [
+                form.order_line.edit(0).name,
+                form.order_line.edit(1).name,
+                form.order_line.edit(2).name,
+                form.sale_order_option_ids.edit(0).name,
+            ]
+
+        order_form = Form(self.sale_order.browse())
+        order_form.sale_order_template_id = self.quotation_template_no_discount
```

Comment:
Hi 👋 

This line is failing without demo data, please fix asap.

https://runbot.odoo.com/web#id=109475&view_type=form&model=runbot.build.error&menu_id=405&cids=1

Reply:
Whoops, didn't realize there was a specific group for sale order templates.

This PR ought to fix it: https://github.com/odoo/odoo/pull/190031